diff --git a/src/inc/utils.h b/src/inc/utils.h
index c0cf9f3..760141d 100644
--- a/src/inc/utils.h
+++ b/src/inc/utils.h
@@ -14,9 +14,19 @@
 #define UTILS_H
 
 #include <stdlib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 
 #define ARRLEN(arr)     (sizeof(arr)/sizeof(*arr))
 
+#ifndef MIN
+#define MIN(a,b) ((a) < (b) ? (a) : (b))
+#endif
+
+#ifndef MAX
+#define MAX(a,b) ((a) > (b) ? (a) : (b))
+#endif
+
 int mkpath(mode_t mode, const char* fmt, ...);
 
 #endif /* UTILS_H */
diff --git a/src/utils.c b/src/utils.c
index 070d9c2..376d36f 100644
--- a/src/utils.c
+++ b/src/utils.c
@@ -9,17 +9,32 @@
 #include <stdio.h>
 #include <stdarg.h>
 
+#ifdef _WIN32
+#include <direct.h>
+#endif
+
 #include "utils.h"
 
 static int _mkpath(char* file_path, mode_t mode) {
     for (char* p = strchr(file_path + 1, '/'); p; p = strchr(p + 1, '/')) {
         *p = '\0';
+
+#ifdef _WIN32
+        if (_mkdir(file_path) == -1) {
+            if (errno != EEXIST) {
+                *p = '/';
+                return -1;
+            }
+        }
+#else
         if (mkdir(file_path, mode) == -1) {
             if (errno != EEXIST) {
                 *p = '/';
                 return -1;
             }
         }
+#endif
+
         *p = '/';
     }
     return 0;
diff --git a/src/v2/imm.c b/src/v2/imm.c
index 3ccc8e7..9817bb6 100644
--- a/src/v2/imm.c
+++ b/src/v2/imm.c
@@ -10,6 +10,7 @@
 
 #include "v2/imm.h"
 #include "log.h"
+#include "utils.h"
 
 void imm_match_free(ImmMatch * iter) {
     while(iter) {
diff --git a/src/v2/imm_finder.c b/src/v2/imm_finder.c
index cf71ae7..8bfbab9 100644
--- a/src/v2/imm_finder.c
+++ b/src/v2/imm_finder.c
@@ -14,6 +14,7 @@
 #include "v2/imm.h"
 #include "v2/inst.h"
 #include "log.h"
+#include "utils.h"
 
 static void imm_append(ImmResult * res, ImmMatch * imm) {
     imm->next = res->matches;
diff --git a/src/v2/strings.c b/src/v2/strings.c
index 43b62fd..a43f425 100644
--- a/src/v2/strings.c
+++ b/src/v2/strings.c
@@ -21,6 +21,47 @@
 #include "v2/strings.h"
 #include "utils.h"
 
+#ifdef _WIN32
+ssize_t getline(char **lineptr, size_t *n, FILE *stream) {
+    if (!lineptr || !n || !stream) return -1;
+
+    char *buf = *lineptr;
+    size_t size = *n;
+    int c = 0;
+    size_t len = 0;
+
+    if (buf == NULL || size == 0) {
+        size = 128;
+        buf = malloc(size);
+        if (!buf) return -1;
+        *lineptr = buf;
+        *n = size;
+    }
+
+    while ((c = fgetc(stream)) != EOF) {
+        if (len + 1 >= size) {
+            size *= 2;
+            char *newbuf = realloc(buf, size);
+            if (!newbuf) return -1;
+            buf = newbuf;
+            *lineptr = buf;
+            *n = size;
+        }
+
+        buf[len++] = c;
+
+        if (c == '\n')
+            break;
+    }
+
+    if (len == 0 && c == EOF)
+        return -1;
+
+    buf[len] = '\0';
+    return len;
+}
+#endif
+
 #define MAX_STRING_LEN      2048
 
 typedef struct _DataRef DataRef;
