diff --git a/Makefile b/Makefile
index 2542399..576e77d 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ include $(DEVKITPRO)/libnx/switch_rules
 APP_TITLE		:=	EdiZon
 APP_FILENAME	:=  ovlEdiZon
 APP_AUTHOR		:=	WerWolv, proferabg, and ppkantorski
-APP_VERSION		:=	v1.0.15
+APP_VERSION		:=	1.0.15
 
 ifeq ($(RELEASE), 1)
 	APP_VERSION	:=	$(APP_VERSION)-$(shell git describe --always)
diff --git a/source/main.cpp b/source/main.cpp
index ea248cc..c76369a 100644
--- a/source/main.cpp
+++ b/source/main.cpp
@@ -48,23 +48,23 @@ public:
     virtual tsl::elm::Element* createUI() {
         auto *rootFrame = new tsl::elm::HeaderOverlayFrame();
         rootFrame->setHeader(new tsl::elm::CustomDrawer([this](tsl::gfx::Renderer *renderer, s32 x, s32 y, s32 w, s32 h) {
-            renderer->drawString(APP_TITLE, false, 20, 50, 32, (tsl::defaultOverlayColor));
+            renderer->drawString("EdiZon", false, 20, 50, 32, (tsl::defaultOverlayColor));
             renderer->drawString(APP_VERSION, false, 20, 52+23, 15, (tsl::bannerVersionTextColor));
 
             if (edz::cheat::CheatManager::getProcessID() != 0) {
-                renderer->drawString("Program ID", false, 150 +14, 40 -6, 15, (tsl::style::color::ColorText));
-                renderer->drawString("Build ID", false, 150 +14, 60 -6, 15, (tsl::style::color::ColorText));
-                renderer->drawString("Process ID", false, 150 +14, 80 -6, 15, (tsl::style::color::ColorText));
-                renderer->drawString(GuiMain::s_runningTitleIDString.c_str(), false, 250 +14, 40 -6, 15, (tsl::style::color::ColorHighlight));
-                renderer->drawString(GuiMain::s_runningBuildIDString.c_str(), false, 250 +14, 60 -6, 15, (tsl::style::color::ColorHighlight));
-                renderer->drawString(GuiMain::s_runningProcessIDString.c_str(), false, 250 +14, 80 -6, 15, (tsl::style::color::ColorHighlight));
+                renderer->drawString("타이틀 ID:", false, 150 +14, 40 -6, 15, (tsl::defaultTextColor));
+                renderer->drawString("빌드 ID:", false, 150 +14, 60 -6, 15, (tsl::defaultTextColor));
+                renderer->drawString("프로세스 ID:", false, 150 +14, 80 -6, 15, (tsl::defaultTextColor));
+                renderer->drawString(GuiMain::s_runningTitleIDString.c_str(), false, 250 +14, 40 -6, 15, (tsl::onTextColor));
+                renderer->drawString(GuiMain::s_runningBuildIDString.c_str(), false, 250 +14, 60 -6, 15, (tsl::onTextColor));
+                renderer->drawString(GuiMain::s_runningProcessIDString.c_str(), false, 250 +14, 80 -6, 15, (tsl::onTextColor));
             }
         }));
 
         auto list = new tsl::elm::List();
 
         if(edz::cheat::CheatManager::isCheatServiceAvailable()){
-            auto cheatsItem = new tsl::elm::ListItem("Cheats");
+            auto cheatsItem = new tsl::elm::ListItem("치트 매니저");
             cheatsItem->setClickListener([cheatsItem](s64 keys) {
                 if (keys & KEY_A) {
                     tsl::shiftItemFocus(cheatsItem);
@@ -75,11 +75,11 @@ public:
             });
             list->addItem(cheatsItem);
         } else {
-            auto noDmntSvc = new tsl::elm::ListItem("Cheat Service Unavailable!");
+            auto noDmntSvc = new tsl::elm::ListItem("치트를 이용할 수 없습니다!");
             list->addItem(noDmntSvc);
         }
 
-        auto statsItem  = new tsl::elm::ListItem("System Information");
+        auto statsItem  = new tsl::elm::ListItem("시스템 정보");
         statsItem->setClickListener([statsItem](s64 keys) {
             if (keys & KEY_A) {
                 tsl::shiftItemFocus(statsItem);
@@ -119,36 +119,36 @@ public:
        // bool setOnce = true; // for ensuring header sync with frame caching for header overlayframe
 
         rootFrame->setHeader(new tsl::elm::CustomDrawer([this](tsl::gfx::Renderer *renderer, s32 x, s32 y, s32 w, s32 h) {
-            renderer->drawString(APP_TITLE, false, 20, 50, 32, (tsl::defaultOverlayColor));
-            renderer->drawString("Cheats", false, 20, 52+23, 15, (tsl::bannerVersionTextColor));
+            renderer->drawString("EdiZon", false, 20, 50, 32, (tsl::defaultOverlayColor));
+            renderer->drawString("치트 매니저", false, 20, 52+23, 15, (tsl::bannerVersionTextColor));
             
 
             if (edz::cheat::CheatManager::getProcessID() != 0) {
-                renderer->drawString("Program ID", false, 150 +14, 40 -6, 15, (tsl::style::color::ColorText));
-                renderer->drawString("Build ID", false, 150 +14, 60 -6, 15, (tsl::style::color::ColorText));
-                renderer->drawString("Process ID", false, 150 +14, 80 -6, 15, (tsl::style::color::ColorText));
-                renderer->drawString(GuiMain::s_runningTitleIDString.c_str(), false, 250 +14, 40 -6, 15, (tsl::style::color::ColorHighlight));
-                renderer->drawString(GuiMain::s_runningBuildIDString.c_str(), false, 250 +14, 60 -6, 15, (tsl::style::color::ColorHighlight));
-                renderer->drawString(GuiMain::s_runningProcessIDString.c_str(), false, 250 +14, 80 -6, 15, (tsl::style::color::ColorHighlight));
+                renderer->drawString("타이틀 ID:", false, 150 +14, 40 -6, 15, (tsl::defaultTextColor));
+                renderer->drawString("빌드 ID:", false, 150 +14, 60 -6, 15, (tsl::defaultTextColor));
+                renderer->drawString("프로세스 ID:", false, 150 +14, 80 -6, 15, (tsl::defaultTextColor));
+                renderer->drawString(GuiMain::s_runningTitleIDString.c_str(), false, 250 +14, 40 -6, 15, (tsl::onTextColor));
+                renderer->drawString(GuiMain::s_runningBuildIDString.c_str(), false, 250 +14, 60 -6, 15, (tsl::onTextColor));
+                renderer->drawString(GuiMain::s_runningProcessIDString.c_str(), false, 250 +14, 80 -6, 15, (tsl::onTextColor));
             }
         }));
 
         if (edz::cheat::CheatManager::getCheats().size() == 0) {
             auto warning = new tsl::elm::CustomDrawer([](tsl::gfx::Renderer *renderer, u16 x, u16 y, u16 w, u16 h){
                 static const auto iconWidth = renderer->getTextDimensions("\uE150", false, 90).first;
-                static const auto textWidth = renderer->getTextDimensions("No Cheats loaded!", false, 25).first;
-                renderer->drawString("\uE150", false, (tsl::cfg::FramebufferWidth - iconWidth) / 2, 274, 90, (0xFFFF));
-                renderer->drawString("No Cheats loaded!", false, (tsl::cfg::FramebufferWidth - textWidth) / 2, 360, 25, (0xFFFF));
+                static const auto textWidth = renderer->getTextDimensions("치트 로드 실패!", false, 25).first;
+                renderer->drawString("\uE150", false, (tsl::cfg::FramebufferWidth - iconWidth) / 2, 274, 90, (tsl::defaultTextColor));
+                renderer->drawString("치트 로드 실패!", false, (tsl::cfg::FramebufferWidth - textWidth) / 2, 360, 25, (tsl::defaultTextColor));
             });
 
             rootFrame->setContent(warning);
 
         } else {
             auto list = new tsl::elm::List();
-            std::string head = "Section: " + this->m_section;
+            std::string head = "섹션: " + this->m_section;
 
             if(m_section.length() > 0) list->addItem(new tsl::elm::CategoryHeader(head));
-            else list->addItem(new tsl::elm::CategoryHeader("Available cheats"));
+            else list->addItem(new tsl::elm::CategoryHeader("이용 가능한 치트"));
 
             bool skip = false, inSection = false, submenus = true;
             std::string skipUntil = "";
@@ -199,7 +199,7 @@ public:
                         replaceAll(cheatNameCheck, ":ENABLED", "");
 
                         auto cheatToggleItem = new tsl::elm::ToggleListItem(/*formatString("%d:%s: %s", cheat->getID(), (cheat->isEnabled() ? "y" : "n"),*/ cheatNameCheck/*.c_str()).c_str()*/, cheat->isEnabled());
-                        cheatToggleItem->setStateChangedListener([&cheat](bool state) { cheat->setState(state);});
+                        cheatToggleItem->setStateChangedListener([&cheat](bool state) { cheat->setState(state); });
 
                         this->m_cheatToggleItems.insert({cheat->getID(), cheatToggleItem});
                         list->addItem(cheatToggleItem);
@@ -227,9 +227,9 @@ public:
             if(this->m_numCheats < 1){
                 auto warning = new tsl::elm::CustomDrawer([](tsl::gfx::Renderer *renderer, u16 x, u16 y, u16 w, u16 h){
                     static const auto iconWidth = renderer->getTextDimensions("\uE150", false, 90).first;
-                    static const auto textWidth = renderer->getTextDimensions("No Cheats in Submenu!", false, 25).first;
-                    renderer->drawString("\uE150", false, (tsl::cfg::FramebufferWidth - iconWidth) / 2, 250, 90, (0xFFFF));
-                    renderer->drawString("No Cheats in Submenu!", false, (tsl::cfg::FramebufferWidth - textWidth) / 2, 340, 25, (0xFFFF));
+                    static const auto textWidth = renderer->getTextDimensions("하위 메뉴에 치트가 없습니다!", false, 25).first;
+                    renderer->drawString("\uE150", false, (tsl::cfg::FramebufferWidth - iconWidth) / 2, 250, 90, (tsl::defaultTextColor));
+                    renderer->drawString("하위 메뉴에 치트가 없습니다!", false, (tsl::cfg::FramebufferWidth - textWidth) / 2, 340, 25, (tsl::defaultTextColor));
                 });
 
                 rootFrame->setContent(warning);
@@ -286,20 +286,20 @@ public:
      }
             
     virtual tsl::elm::Element* createUI() override {
-        auto rootFrame = new tsl::elm::OverlayFrame(APP_TITLE, "System Information");
+        auto rootFrame = new tsl::elm::OverlayFrame("EdiZon", "시스템 정보");
     
         auto infos = new tsl::elm::CustomDrawer([this](tsl::gfx::Renderer *renderer, u16 x, u16 y, u16 w, u16 h){
     
-            renderer->drawString("CPU Temperature:", false, 63, 200, 18, (tsl::style::color::ColorText));
-            renderer->drawString("PCB Temperature:", false, 63, 230, 18, (tsl::style::color::ColorText));
+            renderer->drawString("CPU 온도:", false, 63, 200, 18, (tsl::defaultTextColor));
+            renderer->drawString("PCB 온도:", false, 63, 230, 18, (tsl::defaultTextColor));
     
-            renderer->drawRect(x, 243, w, 1, renderer->a(tsl::style::color::ColorFrame));
-            renderer->drawString("CPU Clock:", false, 63, 270, 18, (tsl::style::color::ColorText));
-            renderer->drawString("GPU Clock:", false, 63, 300, 18, (tsl::style::color::ColorText));
-            renderer->drawString("MEM Clock:", false, 63, 330, 18, (tsl::style::color::ColorText));
+            renderer->drawRect(x, 243, w, 1, renderer->a(tsl::separatorColor));
+            renderer->drawString("CPU 클럭:", false, 63, 270, 18, (tsl::defaultTextColor));
+            renderer->drawString("GPU 클럭:", false, 63, 300, 18, (tsl::defaultTextColor));
+            renderer->drawString("RAM 클럭:", false, 63, 330, 18, (tsl::defaultTextColor));
     
-            renderer->drawRect(x, 343, w, 1, renderer->a(tsl::style::color::ColorFrame));
-            renderer->drawString("Local IP:", false, 63, 370, 18, (tsl::style::color::ColorText));
+            renderer->drawRect(x, 343, w, 1, renderer->a(tsl::separatorColor));
+            renderer->drawString("로컬 IP:", false, 63, 370, 18, (tsl::defaultTextColor));
     
     
             // Draw temperatures and battery percentage
@@ -316,8 +316,8 @@ public:
             snprintf(SOC_temperatureStr, sizeof(SOC_temperatureStr) - 1, "%.1f °C", static_cast<double>(tempSOC));
             snprintf(PCB_temperatureStr, sizeof(PCB_temperatureStr) - 1, "%.1f °C", static_cast<double>(tempPCB));
             
-            renderer->drawString(SOC_temperatureStr, false, 258, 200, 18, (tsl::style::color::ColorHighlight));
-            renderer->drawString(PCB_temperatureStr, false, 258, 230, 18, (tsl::style::color::ColorHighlight));
+            renderer->drawString(SOC_temperatureStr, false, 258, 200, 18, (tsl::onTextColor));
+            renderer->drawString(PCB_temperatureStr, false, 258, 230, 18, (tsl::onTextColor));
             
             static u32 cpuClock = 0, gpuClock = 0, memClock = 0;
     
@@ -331,47 +331,45 @@ public:
                 pcvGetClockRate(PcvModule_EMC, &memClock);
             }
     
-            renderer->drawString(formatString("%.01f MHz", cpuClock / 1'000'000.0F).c_str(), false, 258, 270, 18, (tsl::style::color::ColorHighlight));
-            renderer->drawString(formatString("%.01f MHz", gpuClock / 1'000'000.0F).c_str(), false, 258, 300, 18, (tsl::style::color::ColorHighlight));
-            renderer->drawString(formatString("%.01f MHz", memClock / 1'000'000.0F).c_str(), false, 258, 330, 18, (tsl::style::color::ColorHighlight));
+            renderer->drawString(formatString("%.01f MHz", cpuClock / 1'000'000.0F).c_str(), false, 258, 270, 18, (tsl::onTextColor));
+            renderer->drawString(formatString("%.01f MHz", gpuClock / 1'000'000.0F).c_str(), false, 258, 300, 18, (tsl::onTextColor));
+            renderer->drawString(formatString("%.01f MHz", memClock / 1'000'000.0F).c_str(), false, 258, 330, 18, (tsl::onTextColor));
     
             if (this->m_ipAddressString ==  "0.0.0.0")
-                renderer->drawString("Offline", false, 258, 370, 18, (tsl::style::color::ColorHighlight));
+                renderer->drawString("오프라인", false, 258, 370, 18, (tsl::onTextColor));
             else 
-                renderer->drawString(this->m_ipAddressString.c_str(), false, 258, 370, 18, (tsl::style::color::ColorHighlight));
+                renderer->drawString(this->m_ipAddressString.c_str(), false, 258, 370, 18, (tsl::onTextColor));
     
             if(hosversionAtLeast(15,0,0)){
                 NifmInternetConnectionType conType;
                 u32 wifiStrength;
                 NifmInternetConnectionStatus conStatus;
                 nifmGetInternetConnectionStatus(&conType, &wifiStrength, &conStatus);
-                renderer->drawString("Connection:", false, 63, 400, 18, (tsl::style::color::ColorText));
+                renderer->drawString("연결 상태:", false, 63, 400, 18, (tsl::defaultTextColor));
                 if(conStatus == NifmInternetConnectionStatus_Connected && conType == NifmInternetConnectionType_WiFi) {
-                    std::string wifiStrengthStr = "(Strong)";
-                    tsl::Color color = tsl::Color(0x0, 0xF, 0x0, 0xF);
+                    std::string wifiStrengthStr = "(강함)";
+                    tsl::Color color = renderer->a(tsl::batteryChargingColor);
                     if(wifiStrength == 2){
-                        wifiStrengthStr = "(Fair)";
-                        color = tsl::Color(0xE, 0xE, 0x2, 0xF);
+                        wifiStrengthStr = "(보통)";
+                        color = renderer->a(tsl::starColor);
                     } else if(wifiStrength <= 1){
-                        wifiStrengthStr = "(Poor)";
-                        color = tsl::Color(0xF, 0x0, 0x0, 0xF);
+                        wifiStrengthStr = "(약함)";
+                        color = renderer->a(tsl::warningTextColor);
                     }
-                    renderer->drawString("WiFi", false, 258, 400, 18, (tsl::style::color::ColorHighlight));
+                    renderer->drawString("WiFi", false, 258, 400, 18, (tsl::onTextColor));
                     renderer->drawString(wifiStrengthStr.c_str(), false, 303, 400, 18, (color));
                 } else if(conStatus == NifmInternetConnectionStatus_Connected && conType == NifmInternetConnectionType_Ethernet){
-                    renderer->drawString("Ethernet", false, 258, 400, 18, (tsl::style::color::ColorHighlight));
+                    renderer->drawString("이더넷", false, 258, 400, 18, (tsl::onTextColor));
                 } else {
-                    renderer->drawString("Disconnected", false, 258, 400, 18, (tsl::style::color::ColorHighlight));
+                    renderer->drawString("연결 없음", false, 258, 400, 18, (tsl::onTextColor));
                 }
             } else {
                 s32 signalStrength = 0;
                 wlaninfGetRSSI(&signalStrength);
     
-                renderer->drawString("WiFi Signal:", false, 63, 400, 18, (tsl::style::color::ColorText));
-                renderer->drawString(formatString("%d dBm", signalStrength).c_str(), false, 258, 400, 18, (tsl::style::color::ColorHighlight)); 
+                renderer->drawString("WiFi 신호:", false, 63, 400, 18, (tsl::defaultTextColor));
+                renderer->drawString(formatString("%d dBm", signalStrength).c_str(), false, 258, 400, 18, (tsl::onTextColor)); 
             }
-            renderer->drawString("Credits:", false, 63, 600, 18, (tsl::style::color::ColorText));
-            renderer->drawString(APP_AUTHOR, false, 63, 630, 18, (tsl::style::color::ColorHighlight)); 
         });
         rootFrame->setContent(infos);
     
