diff --git a/Makefile b/Makefile
index ed2ae41..ce77da2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,21 +1,31 @@
 TARGETS := lib/libfancontrol sysmodule overlay
 
-.PHONY: all $(TARGETS)
+.PHONY: all clean $(TARGETS)
+
+all: $(TARGETS)
+	mkdir -p ./out/switch/.overlays
+	cp ./overlay/NX-FanControl.ovl ./out/switch/.overlays/NX-FanControl.ovl
+	
+	mkdir -p ./out/atmosphere/contents/00FF0000B378D640/flags
+	cp ./sysmodule/sysmodule.nsp ./out/atmosphere/contents/00FF0000B378D640/exefs.nsp
+	touch ./out/atmosphere/contents/00FF0000B378D640/flags/boot2.flag
+	@echo "build complete."
 
 lib/libfancontrol:
-	$(MAKE) -C $@
+	$(MAKE) -C $@ -j$(nproc)
 
 overlay:
-	$(MAKE) -C $@
+	$(MAKE) -C $@ -j$(nproc)
 
 sysmodule:
-	$(MAKE) -C $@
+	$(MAKE) -C $@ -j$(nproc)
 
-all: $(TARGETS)
-	mkdir -p ./out/switch/.overlays
-	cp ./overlay/overlay.ovl ./out/switch/.overlays/Nx-FanControl.ovl
-	
-	mkdir -p ./out/atmosphere/contents/00FF0000B378D640/flags
-	cp ./sysmodule/sysmodule.nsp ./out/atmosphere/contents/00FF0000B378D640/exefs.nsp
-	cp ./sysmodule/toolbox.json ./out/atmosphere/contents/00FF0000B378D640/
-	touch ./out/atmosphere/contents/00FF0000B378D640/flags/boot2.flag
\ No newline at end of file
+clean:
+	@rm -rf ./out
+	@rm -rf ./lib/libfancontrol/debug
+	@rm -rf ./lib/libfancontrol/lib
+	@rm -rf ./lib/libfancontrol/release
+	@rm -rf ./sysmodule/build
+	@rm -rf ./overlay/build
+	@find ./sysmodule -type f \( -name '*.elf' -o -name '*.npdm' -o -name '*.nso' -o -name '*.nsp' \) -exec rm -f {} \;
+	@find ./overlay -type f \( -name '*.elf' -o -name '*.nacp' -o -name '*.ovl' \) -exec rm -f {} \;
\ No newline at end of file
diff --git a/overlay/Makefile b/overlay/Makefile
index c8a6529..47cdb3e 100644
--- a/overlay/Makefile
+++ b/overlay/Makefile
@@ -38,7 +38,7 @@ include $(DEVKITPRO)/libnx/switch_rules
 #   NACP building is skipped as well.
 #---------------------------------------------------------------------------------
 APP_TITLE	:=	NX-FanControl
-APP_VERSION :=	1.0.3+r4
+APP_VERSION :=	1.0.4+r1
 
 TARGET		:=	$(APP_TITLE)
 BUILD		:=	build
@@ -65,7 +65,7 @@ CFLAGS := -g -Wall -O3 -ffunction-sections -fdata-sections -flto \
 CFLAGS += $(INCLUDE) -D__SWITCH__ -DAPP_VERSION="\"$(APP_VERSION)\"" -D_FORTIFY_SOURCE=2
 
 # Enable appearance overriding
-UI_OVERRIDE_PATH := /config/NX-FanControl/
+UI_OVERRIDE_PATH := /config/ultrahand/
 CFLAGS += -DUI_OVERRIDE_PATH="\"$(UI_OVERRIDE_PATH)\""
 
 # For theme / wallpaper loading in GUI class method (add to project if theme does not appear)
diff --git a/overlay/source/main_menu.cpp b/overlay/source/main_menu.cpp
index 81b1e35..a02fa1a 100644
--- a/overlay/source/main_menu.cpp
+++ b/overlay/source/main_menu.cpp
@@ -13,11 +13,11 @@ MainMenu::MainMenu()
 
     if (IsRunning() != 0)
     {
-        this->_enabledBtn = new tsl::elm::ToggleListItem("Enabled", true);
+        this->_enabledBtn = new tsl::elm::ToggleListItem("활성화", true);
     }
     else
     {
-        this->_enabledBtn = new tsl::elm::ToggleListItem("Enabled", false);
+        this->_enabledBtn = new tsl::elm::ToggleListItem("활성화", false);
     }
 }
 
@@ -27,6 +27,7 @@ tsl::elm::Element* MainMenu::createUI()
 
     auto list = new tsl::elm::List();
 
+    list->addItem(new tsl::elm::CategoryHeader("팬 쿨링", true));
     this->_enabledBtn->setStateChangedListener([this](bool state)
     {
         if (state)
@@ -50,7 +51,7 @@ tsl::elm::Element* MainMenu::createUI()
     });
     list->addItem(this->_enabledBtn);
 
-    list->addItem(new tsl::elm::CategoryHeader("Fan Curve", true));
+    list->addItem(new tsl::elm::CategoryHeader("팬 커브", true));
     this->_p0Label->setClickListener([this](uint64_t keys)
     {
         if (keys & HidNpadButton_A) 
diff --git a/overlay/source/select_menu.cpp b/overlay/source/select_menu.cpp
index d223166..938c87c 100644
--- a/overlay/source/select_menu.cpp
+++ b/overlay/source/select_menu.cpp
@@ -7,7 +7,7 @@ SelectMenu::SelectMenu(int i, TemperaturePoint* fanCurveTable, bool* tableIsChan
     this->_fanCurveTable = fanCurveTable;
     this->_tableIsChanged = tableIsChanged;
 
-    this->_saveBtn = new tsl::elm::ListItem("Save");
+    this->_saveBtn = new tsl::elm::ListItem("저장");
     this->_tempLabel = new tsl::elm::CategoryHeader(std::to_string((this->_fanCurveTable + this->_i)->temperature_c) + "°C", true);
     this->_fanLabel = new tsl::elm::CategoryHeader(std::to_string((int)((this->_fanCurveTable + this->_i)->fanLevel_f * 100)) + "%", true);
 }
@@ -22,7 +22,7 @@ tsl::elm::Element* SelectMenu::createUI(){
     {
         this->_tempLabel->setText(std::to_string(value) + "°C");
         (this->_fanCurveTable + this->_i)->temperature_c = value;
-        this->_saveBtn->setText("Save");
+        this->_saveBtn->setText("저장");
     });
     trackTemp->setProgress((this->_fanCurveTable + this->_i)->temperature_c);
     list->addItem(trackTemp);
@@ -33,7 +33,7 @@ tsl::elm::Element* SelectMenu::createUI(){
     {
         this->_fanLabel->setText(std::to_string(value) + "%");
         (this->_fanCurveTable + this->_i)->fanLevel_f = (float)value / 100;
-        this->_saveBtn->setText("Save");
+        this->_saveBtn->setText("저장");
     });
     trackFanL->setProgress((int)((this->_fanCurveTable + this->_i)->fanLevel_f * 100));
     list->addItem(trackFanL);
@@ -58,7 +58,7 @@ tsl::elm::Element* SelectMenu::createUI(){
                 pmshellLaunchProgram(0, &programLocation, &pid);
             }
                 
-            this->_saveBtn->setText("Saved!");
+            this->_saveBtn->setText("설정 저장됨!");
             *this->_tableIsChanged = true;
             return true;
         }
