diff --git a/Overlay/source/main.cpp b/Overlay/source/main.cpp
index ad3136c..82b17d7 100644
--- a/Overlay/source/main.cpp
+++ b/Overlay/source/main.cpp
@@ -146,12 +146,12 @@ public:
 	virtual tsl::elm::Element* createUI() override {
 		// A OverlayFrame is the base element every overlay consists of. This will draw the default Title and Subtitle.
 		// If you need more information in the header or want to change it's look, use a HeaderOverlayFrame.
-		auto frame = new tsl::elm::OverlayFrame("ReverseNX-RT", _isDocked ? "Change Docked Default Display Resolution" : "Change Handheld Default Display Resolution");
+		auto frame = new tsl::elm::OverlayFrame("ReverseNX-RT", _isDocked ? "독 모드 기본 해상도로 변경" : "휴대 모드 기본 해상도로 변경");
 
 		// A list that can contain sub elements and handles scrolling
 		auto list = new tsl::elm::List();
 
-		auto *clickableListItem2 = new tsl::elm::ListItem("Default");
+		auto *clickableListItem2 = new tsl::elm::ListItem("기본");
 		clickableListItem2->setClickListener([this](u64 keys) { 
 			if ((keys & HidNpadButton_A) && PluginRunning) {
 				if (_isDocked) ReverseNX_RT->res.docked_res = res_mode_default;
@@ -212,44 +212,44 @@ public:
 		
 		list->addItem(new tsl::elm::CustomDrawer([](tsl::gfx::Renderer *renderer, s32 x, s32 y, s32 w, s32 h) {
 			if (!SaltySD) {
-				renderer->drawString("SaltyNX is not working!", false, x+7, y+50, 20, renderer->a(0xF33F));
+				renderer->drawString("SaltyNX가 작동하지 않습니다!", false, x+7, y+50, 20, renderer->a(tsl::warningTextColor));
 			}
 			else if (!check) {
 				if (closed) {
-					renderer->drawString("Game was closed! Overlay disabled!", false, x+7, y+20, 19, renderer->a(0xF33F));
+					renderer->drawString("타이틀 종료 됨, 오버레이 비활성화!", false, x+7, y+20, 19, renderer->a(tsl::warningTextColor));
 				}
 				else {
-					renderer->drawString("Game is not running! Overlay disabled!", false, x+7, y+20, 19, renderer->a(0xF33F));
+					renderer->drawString("실행 타이틀 없음, 오버레이 비활성화!", false, x+7, y+20, 19, renderer->a(tsl::warningTextColor));
 				}
 			}
 			else if (!PluginRunning) {
-				renderer->drawString("Game is running.", false, x+7, y+20, 20, renderer->a(0xFFFF));
-				renderer->drawString("ReverseNX-RT is not running!", false, x+7, y+40, 20, renderer->a(0xF33F));
+				renderer->drawString("타이틀 실행중", false, x+7, y+20, 20, renderer->a(tsl::sectionTextColor));
+				renderer->drawString("ReverseNX-RT가 작동하지 않습니다!", false, x+7, y+40, 20, renderer->a(tsl::warningTextColor));
 			}
 			else {
-				renderer->drawString("ReverseNX-RT is running.", false, x+7, y+20, 20, renderer->a(0xFFFF));
-				if (!(ReverseNX_RT->pluginActive)) renderer->drawString("Game didn't check any mode!", false, x+7, y+40, 18, renderer->a(0xF33F));
+				renderer->drawString("ReverseNX-RT 실행중", false, x+7, y+20, 20, renderer->a(tsl::sectionTextColor));
+				if (!(ReverseNX_RT->pluginActive)) renderer->drawString("모드를 확인할 수 없습니다!", false, x+7, y+40, 18, renderer->a(tsl::warningTextColor));
 				else {
-					renderer->drawString(SystemChar, false, x+7, y+42, 20, renderer->a(0xFFFF));
-					renderer->drawString(DockedChar, false, x+7, y+64, 20, renderer->a(0xFFFF));
+					renderer->drawString(SystemChar, false, x+7, y+42, 20, renderer->a(tsl::sectionTextColor));
+					renderer->drawString(DockedChar, false, x+7, y+64, 20, renderer->a(tsl::sectionTextColor));
 					if (!(ReverseNX_RT->def)) {
 						if (ReverseNX_RT->wasDDRused) {
-							renderer->drawString(HandheldDDR, false, x+7, y+86, 20, renderer->a(0xFFFF));
-							renderer->drawString(DockedDDR, false, x+7, y+108, 20, renderer->a(0xFFFF));
+							renderer->drawString(HandheldDDR, false, x+7, y+86, 20, renderer->a(tsl::sectionTextColor));
+							renderer->drawString(DockedDDR, false, x+7, y+108, 20, renderer->a(tsl::sectionTextColor));
 						}
 						else {
-							renderer->drawString("Default Display Resolution", false, x+7, y+86, 20, renderer->a(0xFFFF));
-							renderer->drawString("was not checked!", false, x+7, y+108, 20, renderer->a(0xFFFF));							
+							renderer->drawString("기본 해상도", false, x+7, y+86, 20, renderer->a(tsl::sectionTextColor));
+							renderer->drawString("확인할 수 없습니다!", false, x+7, y+108, 20, renderer->a(tsl::sectionTextColor));
 						}
 					}
 				}
-				renderer->drawString(saveChar, false, x+7, y+130, 20, renderer->a(0xFFFF));
+				renderer->drawString(saveChar, false, x+7, y+130, 20, renderer->a(tsl::sectionTextColor));
 			}
 	}), 150);
 
 		if (PluginRunning && ReverseNX_RT->pluginActive) {
 
-			auto *clickableListItem = new tsl::elm::ListItem("Change system control");
+			auto *clickableListItem = new tsl::elm::ListItem("시스템 제어");
 			clickableListItem->setClickListener([](u64 keys) { 
 				if ((keys & HidNpadButton_A) && PluginRunning) {
 					ReverseNX_RT->def = !(ReverseNX_RT->def);
@@ -264,7 +264,7 @@ public:
 
 			if (!(ReverseNX_RT->def)) {
 
-				auto *clickableListItem2 = new tsl::elm::ListItem("Change mode");
+				auto *clickableListItem2 = new tsl::elm::ListItem("모드 변경");
 				clickableListItem2->setClickListener([](u64 keys) { 
 					if ((keys & HidNpadButton_A) && PluginRunning) {
 						ReverseNX_RT->isDocked = !(ReverseNX_RT->isDocked);
@@ -276,7 +276,7 @@ public:
 				list->addItem(clickableListItem2);
 
 				if (ReverseNX_RT->wasDDRused) {
-					auto *clickableListItem3 = new tsl::elm::ListItem("Change Handheld DDR");
+					auto *clickableListItem3 = new tsl::elm::ListItem("휴대 모드 DDR 변경");
 					clickableListItem3->setClickListener([](u64 keys) { 
 						if ((keys & HidNpadButton_A) && PluginRunning) {
 							tsl::changeTo<ResolutionModeMenu>(false);
@@ -287,7 +287,7 @@ public:
 					});
 					list->addItem(clickableListItem3);
 
-					auto *clickableListItem4 = new tsl::elm::ListItem("Change Docked DDR");
+					auto *clickableListItem4 = new tsl::elm::ListItem("독 모드 DDR 변경");
 					clickableListItem4->setClickListener([](u64 keys) { 
 						if ((keys & HidNpadButton_A) && PluginRunning) {
 							tsl::changeTo<ResolutionModeMenu>(true);
@@ -300,12 +300,12 @@ public:
 				}
 			}
 
-			auto *clickableListItem3 = new tsl::elm::ListItem("Save current settings");
+			auto *clickableListItem3 = new tsl::elm::ListItem("현재 설정을 저장");
 			clickableListItem3->setClickListener([](u64 keys) { 
 				if ((keys & HidNpadButton_A) && PluginRunning) {
 					if (writeSave())
-						snprintf(saveChar, sizeof(saveChar), "Settings saved successfully!");
-					else snprintf(saveChar, sizeof(saveChar), "Saving settings failed!");
+						snprintf(saveChar, sizeof(saveChar), "설정 저장 성공!");
+					else snprintf(saveChar, sizeof(saveChar), "설정 저장 실패!");
 					return true;
 				}
 				
@@ -337,22 +337,22 @@ public:
 				_isDocked = ReverseNX_RT->isDocked;
 				i = 0;
 				
-				if (_def) sprintf(SystemChar, "Controlled by system: Yes");
-				else sprintf(SystemChar, "Controlled by system: No");
+				if (_def) sprintf(SystemChar, "시스템 제어 상태: ON");
+				else sprintf(SystemChar, "시스템 제어 상태: OFF");
 
 				if (_def) {
-					if (_isDocked) sprintf(DockedChar, "Mode: Docked");
-					else sprintf(DockedChar, "Mode: Handheld");
+					if (_isDocked) sprintf(DockedChar, "타입: 독 모드");
+					else sprintf(DockedChar, "타입: 휴대 모드");
 				}
 				else {
-					if (_isDocked) sprintf(DockedChar, "Mode: Fake Docked");
-					else sprintf(DockedChar, "Mode: Fake Handheld");
+					if (_isDocked) sprintf(DockedChar, "타입: 페이크 독 모드");
+					else sprintf(DockedChar, "타입: 페이크 휴대 모드");
 				}
 
-				if (!ReverseNX_RT->res.handheld_res) strcpy(HandheldDDR, "Handheld DDR: Default");
-				else snprintf(HandheldDDR, sizeof(HandheldDDR), "Handheld DDR: %dx%d", resolutions[ReverseNX_RT->res.handheld_res].first, resolutions[ReverseNX_RT->res.handheld_res].second);
-				if (!ReverseNX_RT->res.docked_res) strcpy(DockedDDR, "Docked DDR: Default");
-				else snprintf(DockedDDR, sizeof(DockedDDR), "Docked DDR: %dx%d", resolutions[ReverseNX_RT->res.docked_res].first, resolutions[ReverseNX_RT->res.docked_res].second);
+				if (!ReverseNX_RT->res.handheld_res) strcpy(HandheldDDR, "휴대 모드 DDR: 기본");
+				else snprintf(HandheldDDR, sizeof(HandheldDDR), "휴대 모드 DDR: %dx%d", resolutions[ReverseNX_RT->res.handheld_res].first, resolutions[ReverseNX_RT->res.handheld_res].second);
+				if (!ReverseNX_RT->res.docked_res) strcpy(DockedDDR, "독 모드 DDR: 기본");
+				else snprintf(DockedDDR, sizeof(DockedDDR), "독 모드 DDR: %dx%d", resolutions[ReverseNX_RT->res.docked_res].first, resolutions[ReverseNX_RT->res.docked_res].second);
 			}
 			else i++;
 		}
