diff --git a/Makefile b/Makefile
index 567d756..d263c31 100644
--- a/Makefile
+++ b/Makefile
@@ -15,19 +15,16 @@ export GIT_REVISION := $(GIT_BRANCH)-$(shell git rev-parse --short HEAD)
 export VERSION_DIRTY := $(VERSION)
 export VERSION_WITH_HASH := $(VERSION)-$(shell git rev-parse --short HEAD)
 else
-export GIT_REVISION := $(GIT_BRANCH)-$(shell git rev-parse --short HEAD)-dirty
-export VERSION_DIRTY := $(VERSION)-dirty
-export VERSION_WITH_HASH := $(VERSION)-$(shell git rev-parse --short HEAD)-dirty
+export GIT_REVISION := $(GIT_BRANCH)-$(shell git rev-parse --short HEAD)
+export VERSION_DIRTY := $(VERSION)
+export VERSION_WITH_HASH := $(VERSION)-$(shell git rev-parse --short HEAD)
 endif
 
 export BUILD_DATE := -DDATE_YEAR=\"$(shell date +%Y)\" \
 					-DDATE_MONTH=\"$(shell date +%m)\" \
 					-DDATE_DAY=\"$(shell date +%d)\" \
-					-DDATE_HOUR=\"$(shell date +%H)\" \
-					-DDATE_MIN=\"$(shell date +%M)\" \
-					-DDATE_SEC=\"$(shell date +%S)\" \
 
-export CUSTOM_DEFINES := -DVERSION=\"v$(VERSION)\" \
+export CUSTOM_DEFINES := -DVERSION=\"$(VERSION)\" \
 					-DGIT_BRANCH=\"$(GIT_BRANCH)\" \
 					-DGIT_REVISION=\"$(GIT_REVISION)\" \
 					-DVERSION_DIRTY=\"$(VERSION_DIRTY)\" \
@@ -47,6 +44,9 @@ $(TARGETS):
 clean:
 	@rm -rf out
 	@rm -f sys-patch.zip
+	@rm -rf overlay/{build,out} sysmod/{build,out}
+	@find sysmod/ -type f \( -name '*.elf' -o -name '*.nacp' -o -name '*.ovl' \) -delete
+	@find overlay/ -type f \( -name '*.elf' -o -name '*.npdm' -o -name '*.nso' -o -name '*.nsp' \) -delete
 	@for i in $(TARGETS); do $(MAKE) -C $$i clean || exit 1; done;
 
 dist: all
diff --git a/overlay/Makefile b/overlay/Makefile
index 382a67d..3c3484c 100644
--- a/overlay/Makefile
+++ b/overlay/Makefile
@@ -45,10 +45,11 @@ TARGET		:=	sys-patch-overlay
 BUILD		:=	build
 SOURCES		:=	src ../common/minIni
 DATA		:=	data
-INCLUDES	:=	include ../common libtesla/include
+INCLUDES	:=	include ../common
 
 NO_ICON		:=  1
 
+include ${TOPDIR}/libultrahand/ultrahand.mk
 #---------------------------------------------------------------------------------
 # options for code generation
 #---------------------------------------------------------------------------------
diff --git a/overlay/src/main.cpp b/overlay/src/main.cpp
index 9a7887e..fa42d20 100644
--- a/overlay/src/main.cpp
+++ b/overlay/src/main.cpp
@@ -6,8 +6,8 @@
 
 namespace {
 
-constexpr auto CONFIG_PATH = "/config/sys-patch/config.ini";
-constexpr auto LOG_PATH = "/config/sys-patch/log.ini";
+constexpr auto CONFIG_PATH = "/atmosphere/config/sys-patch_cfg.ini";
+constexpr auto LOG_PATH = "/atmosphere/logs/sys-patch.log";
 
 auto does_file_exist(const char* path) -> bool {
     Result rc{};
@@ -71,32 +71,14 @@ public:
     GuiOptions() { }
 
     tsl::elm::Element* createUI() override {
-        auto frame = new tsl::elm::OverlayFrame("sys-patch", VERSION_WITH_HASH);
+        auto frame = new tsl::elm::OverlayFrame("시스템 패치", VERSION_DIRTY);
         auto list = new tsl::elm::List();
 
-        list->addItem(new tsl::elm::CategoryHeader("Options"));
-        list->addItem(config_patch_sysmmc.create_list_item("Patch sysMMC"));
-        list->addItem(config_patch_emummc.create_list_item("Patch emuMMC"));
-        list->addItem(config_logging.create_list_item("Logging"));
-        list->addItem(config_version_skip.create_list_item("Version skip"));
-
-        frame->setContent(list);
-        return frame;
-    }
-
-    ConfigEntry config_patch_sysmmc{"options", "patch_sysmmc", true};
-    ConfigEntry config_patch_emummc{"options", "patch_emummc", true};
-    ConfigEntry config_logging{"options", "enable_logging", true};
-    ConfigEntry config_version_skip{"options", "version_skip", true};
-};
-
-class GuiToggle final : public tsl::Gui {
-public:
-    GuiToggle() { }
-
-    tsl::elm::Element* createUI() override {
-        auto frame = new tsl::elm::OverlayFrame("sys-patch", VERSION_WITH_HASH);
-        auto list = new tsl::elm::List();
+        list->addItem(new tsl::elm::CategoryHeader("설정"));
+        list->addItem(config_patch_sysmmc.create_list_item("시스낸드 패치"));
+        list->addItem(config_patch_emummc.create_list_item("에뮤낸드 패치"));
+        list->addItem(config_logging.create_list_item("로깅"));
+        list->addItem(config_version_skip.create_list_item("버전 스킵"));
 
         list->addItem(new tsl::elm::CategoryHeader("FS - 0100000000000000"));
         list->addItem(config_noacidsigchk1.create_list_item("noacidsigchk_1.0.0-9.2.0"));
@@ -140,6 +122,11 @@ public:
         return frame;
     }
 
+    ConfigEntry config_patch_sysmmc{"options", "patch_sysmmc", true};
+    ConfigEntry config_patch_emummc{"options", "patch_emummc", true};
+    ConfigEntry config_logging{"options", "enable_logging", true};
+    ConfigEntry config_version_skip{"options", "version_skip", true};
+
     ConfigEntry config_noacidsigchk1{"fs", "noacidsigchk_1.0.0-9.2.0", true};
     ConfigEntry config_noacidsigchk2{"fs", "noacidsigchk_1.0.0-9.2.0", true};
     ConfigEntry config_noncasigchk1{"fs", "noncasigchk_1.0.0-3.0.2", true};
@@ -171,7 +158,7 @@ public:
     GuiLog() { }
 
     tsl::elm::Element* createUI() override {
-        auto frame = new tsl::elm::OverlayFrame("sys-patch", VERSION_WITH_HASH);
+        auto frame = new tsl::elm::OverlayFrame("시스템 패치", VERSION_DIRTY);
         auto list = new tsl::elm::List();
 
         if (does_file_exist(LOG_PATH)) {
@@ -190,37 +177,43 @@ public:
 
                 if (user->last_section != Section) {
                     user->last_section = Section;
-                    user->list->addItem(new tsl::elm::CategoryHeader("Log: " + user->last_section));
+                    user->list->addItem(new tsl::elm::CategoryHeader("로그: " + user->last_section));
                 }
 
-                #define F(x) ((x) >> 4) // 8bit -> 4bit
-                constexpr tsl::Color colour_syspatch{F(0), F(255), F(200), F(255)};
-                constexpr tsl::Color colour_file{F(255), F(177), F(66), F(255)};
-                constexpr tsl::Color colour_unpatched{F(250), F(90), F(58), F(255)};
-                #undef F
-
                 if (value.starts_with("Patched")) {
-                    auto *item = new tsl::elm::ListItem(Key);
-                    item->setValue("Patched", true);
-                    user->list->addItem(item);
-                } else if (value.starts_with("Unpatched") || value.starts_with("Disabled")) {
-                    auto *item = new tsl::elm::ListItem(Key);
-                    item->setValue(Value, true);
-                    user->list->addItem(item);
+                    if (value.ends_with("(sys-patch)")) {
+                        auto li = new tsl::elm::ListItem(Key, "\uE14B");
+                        li->setValueColor(tsl::onTextColor);
+                        user->list->addItem(li);
+                    } else {
+                        auto li = new tsl::elm::ListItem(Key, "\uE14B");
+                        li->setValueColor(tsl::starColor);
+                        user->list->addItem(li);
+                    }
+                } else if (value.starts_with("Unpatched")) {
+                    auto li = new tsl::elm::ListItem(Key, "\uE14E");
+                    li->setValueColor(tsl::warningTextColor);
+                    user->list->addItem(li);
+                } else if (value.starts_with("Disabled")) {
+                    auto li = new tsl::elm::ListItem(Key, "\uE14C");
+                    li->setValueColor(tsl::warningTextColor);
+                    user->list->addItem(li);
                 } else if (user->last_section == "stats") {
-                    auto *item = new tsl::elm::ListItem(Key);
-                    item->setValue(Value, true);
-                    user->list->addItem(item);
+                    auto li = new tsl::elm::ListItem(Key, Value);
+                    li->setValueColor(tsl::bannerVersionTextColor);
+                    user->list->addItem(li);
                 } else {
-                    auto *item = new tsl::elm::ListItem(Key);
-                    item->setValue(Value, true);
-                    user->list->addItem(item);
+                    auto li = new tsl::elm::ListItem(Key, Value);
+                    li->setValueColor(tsl::bannerVersionTextColor);
+                    user->list->addItem(li);
                 }
 
                 return 1;
             }, &callback_userdata, LOG_PATH);
         } else {
-            list->addItem(new tsl::elm::ListItem("No log found!"));
+            auto li = new tsl::elm::ListItem("Atmosphère 패치", "\uE12C");
+            li->setValueColor(tsl::onTextColor);
+            list->addItem(li);
         }
 
         frame->setContent(list);
@@ -233,12 +226,11 @@ public:
     GuiMain() { }
 
     tsl::elm::Element* createUI() override {
-        auto frame = new tsl::elm::OverlayFrame("sys-patch", VERSION_WITH_HASH);
+        auto frame = new tsl::elm::OverlayFrame("시스템 패치", VERSION_DIRTY);
         auto list = new tsl::elm::List();
 
-        auto options = new tsl::elm::ListItem("Options");
-        auto toggle = new tsl::elm::ListItem("Toggle patches");
-        auto log = new tsl::elm::ListItem("Log");
+        auto options = new tsl::elm::ListItem("설정");
+        auto log = new tsl::elm::ListItem("상태");
 
         options->setClickListener([](u64 keys) -> bool {
             if (keys & HidNpadButton_A) {
@@ -248,14 +240,6 @@ public:
             return false;
         });
 
-        toggle->setClickListener([](u64 keys) -> bool {
-            if (keys & HidNpadButton_A) {
-                tsl::changeTo<GuiToggle>();
-                return true;
-            }
-            return false;
-        });
-
         log->setClickListener([](u64 keys) -> bool {
             if (keys & HidNpadButton_A) {
                 tsl::changeTo<GuiLog>();
@@ -264,11 +248,36 @@ public:
             return false;
         });
 
-        list->addItem(new tsl::elm::CategoryHeader("Menu"));
+        list->addItem(new tsl::elm::CategoryHeader("메뉴"));
         list->addItem(options);
-        list->addItem(toggle);
         list->addItem(log);
 
+        list->addItem(new tsl::elm::CategoryHeader("상태: sys-patch 비활성화 시"));
+        
+        list->addItem(new tsl::elm::CustomDrawer([&](tsl::gfx::Renderer *renderer, s32 x, s32 y, s32 w, s32 h) {
+            renderer->drawString("\uE12C", false, x + 15, y + 25, 16, tsl::onTextColor);
+            renderer->drawString("Atmosphère에서 패턴이 패치되었습니다.", false, x + 43, y + 25, 16, renderer->a(tsl::defaultTextColor));
+            renderer->drawString("별도의 패치 파일 없이 플레이 가능합니다.", false, x + 43, y + 45, 16, renderer->a(tsl::defaultTextColor));
+        }), 50);
+
+        list->addItem(new tsl::elm::CategoryHeader("상태: OC Toolkit, loader.kip 활성화 시"));
+        
+        list->addItem(new tsl::elm::CustomDrawer([&](tsl::gfx::Renderer *renderer, s32 x, s32 y, s32 w, s32 h) {
+            renderer->drawString("\uE14B", false, x + 15, y + 25, 16, tsl::onTextColor);
+            renderer->drawString("sys-patch로 패턴이 패치되었습니다.", false, x + 43, y + 25, 16, renderer->a(tsl::defaultTextColor));
+
+            renderer->drawString("\uE14B", false, x + 15, y + 60, 16, tsl::starColor);
+            renderer->drawString("빌트인 패치로 패턴이 패치되었습니다.", false, x + 43, y + 60, 16, renderer->a(tsl::defaultTextColor));
+            
+            renderer->drawString("\uE14E", false, x + 15, y + 95, 16, tsl::warningTextColor);
+            renderer->drawString("대응 패턴이 없거나, 패치할 필요가 없습니다.", false, x + 43, y + 95, 16, renderer->a(tsl::defaultTextColor));
+
+            renderer->drawString("\uE14C", false, x + 15, y + 130, 16, tsl::warningTextColor);
+            renderer->drawString("설정의 활성 상태를 확인하세요.", false, x + 43, y + 130, 16, renderer->a(tsl::defaultTextColor));
+            renderer->drawString("현재 펌웨어 코드의 패턴을 찾을 수 없거나,", false, x + 43, y + 150, 16, renderer->a(tsl::defaultTextColor));
+            renderer->drawString("패치에 실패하였습니다.", false, x + 43, y + 170, 16, renderer->a(tsl::defaultTextColor));
+        }), 175);
+
         frame->setContent(list);
         return frame;
     }
@@ -285,7 +294,7 @@ public:
 } // namespace
 
 int main(int argc, char **argv) {
-    create_dir("/config/");
-    create_dir("/config/sys-patch/");
+    create_dir("/atmosphere/config/");
+    create_dir("/atmosphere/logs/");
     return tsl::loop<SysPatchOverlay>(argc, argv);
 }
diff --git a/sysmod/Makefile b/sysmod/Makefile
index 536873f..e66d977 100644
--- a/sysmod/Makefile
+++ b/sysmod/Makefile
@@ -169,7 +169,6 @@ $(BUILD):
 	@rm -rf out/
 	@mkdir -p out/atmosphere/contents/420000000000000B/flags
 	@touch out/atmosphere/contents/420000000000000B/flags/boot2.flag
-	@cp $(CURDIR)/toolbox.json out/atmosphere/contents/420000000000000B/toolbox.json
 	@cp $(CURDIR)/$(TARGET).nsp out/atmosphere/contents/420000000000000B/exefs.nsp
 
 #---------------------------------------------------------------------------------
diff --git a/sysmod/src/main.cpp b/sysmod/src/main.cpp
index 01edb31..8eff51f 100644
--- a/sysmod/src/main.cpp
+++ b/sysmod/src/main.cpp
@@ -583,11 +583,11 @@ void keygen_to_str(char* s, u8 keygen) {
 } // namespace
 
 int main(int argc, char* argv[]) {
-    constexpr auto ini_path = "/config/sys-patch/config.ini";
-    constexpr auto log_path = "/config/sys-patch/log.ini";
+    constexpr auto ini_path = "/atmosphere/config/sys-patch_config.ini";
+    constexpr auto log_path = "/atmosphere/logs/sys-patch.log";
 
-    create_dir("/config/");
-    create_dir("/config/sys-patch/");
+    create_dir("/atmosphere/config/");
+    create_dir("/atmosphere/logs/");
     ini_remove(log_path);
 
     // load options
@@ -662,7 +662,7 @@ int main(int argc, char* argv[]) {
         ms_2_str(patch_time, diff_ns/1000ULL/1000ULL);
 
         // defined in the Makefile
-        #define DATE (DATE_DAY "." DATE_MONTH "." DATE_YEAR " " DATE_HOUR ":" DATE_MIN ":" DATE_SEC)
+        #define DATE (DATE_YEAR "." DATE_MONTH "." DATE_DAY)
 
         ini_puts("stats", "version", VERSION_WITH_HASH, log_path);
         ini_puts("stats", "build_date", DATE, log_path);
